src/runtime/markdown-parser/handler/utils.ts

```diff
@@ -20,7 +20,7 @@ export function parseThematicBlock (lang: string) {
 
   const language = lang.replace(/[{|[](.+)/, '').match(/^[^ \t]+(?=[ \t]|$)/)
   const highlightTokens = lang.match(/{([^}]*)}/)
-  const filenameTokens = lang.match(/\[([^\]]*)\]/)
+  const filenameTokens = lang.match(/\[(.*)\]/) /** Allow ']' in filename. @see https://regex101.com/r/Hm0t5m/1 */
```

Comment:
I think we used `[^\]]` to allow using lists in the meta part.
Here's an example what would not work anymore if we match the longest string.
> \```language [filename]{hightlights} meta=[...]

It would match the content `filename]{hightlights} meta=[...`

Reply:
I thought we could use something like `\[` in the markdown and prevent matching those somehow

Reply:
Oooh I understand. I do it too quickly!

Reply:
I think, that this might work: `/\[((\\]|[^\]])*)\]/`

---

This part `(\\]|[^\]])*` matches one or a sequence of: `\]` or characters that are not `]`

Only if our string has no more `]` after a `\]` it terminates the match at a `\]`.

e.g.

```
```ts [[...slug\].vue] meta=[]
```

=> `[...slug\].vue` as the first group match. Then we only have to replace `\]` which in reality is `\\]` with `]`-es

Reply:
Yes but if there is no meta, this regex is not working

https://regex101.com/r/bG0Net/1